src/main/webapp/js/media_manager.js

```diff
@@ -1269,14 +1269,22 @@ export class MediaManager {
      * @param {*} period : measurement period
      */
     enableAudioLevelForLocalStream(levelCallback, period) {
-        this.localStreamSoundMeter = new SoundMeter(this.audioContext);
-        this.connectSoundMeterToLocalStream();
-
-        this.soundLevelProviderId = setInterval(() => {
-            levelCallback(this.localStreamSoundMeter.instant.toFixed(2));
-        }, period);
+        this.startAudio(this.audioContext, levelCallback).then(() =>{
+            this.audioContext.resume().then(r => {});
+        });
     }
 
+    async startAudio (context, levelCallback) {
+        await context.audioWorklet.addModule(new URL('./volume-meter-processor.js', import.meta.url)).catch((err) => console.log(err));
+        const mediaStream = await navigator.mediaDevices.getUserMedia({audio: true});
```

Comment:
I think we already have the audio stream. Why are we getting the audio again? It may cause some problems if a specific audio device is selected.